src/h5web/providers/mock/api.ts

```diff
@@ -0,0 +1,77 @@
+import ndarray from 'ndarray';
+import unpack from 'ndarray-unpack';
+import { assertArrayShape, assertPrintableType } from '../../guards';
+import { GetValueParams, ProviderApi } from '../context';
+import type { Entity } from '../models';
+import { mockFilepath } from './metadata';
+import { assertMockDataset, findMockEntity } from './utils';
+
+const SLOW_TIMEOUT = 3000;
+const SLOW_METADATA_PATH = '/resilience/slow_metadata';
+const SLOW_VALUE_PATH = '/resilience/slow_value';
+const ERROR_PATH = '/resilience/error_value';
+
+export class MockApi extends ProviderApi {
+  public constructor() {
+    super(mockFilepath);
+  }
+
+  public async getEntity(path: string): Promise<Entity> {
+    if (path === SLOW_METADATA_PATH) {
+      await new Promise<void>((resolve) => {
+        setTimeout(() => resolve(), SLOW_TIMEOUT);
+      });
+    }
+
+    return findMockEntity(path);
+  }
+
+  public async getValue(params: GetValueParams): Promise<unknown> {
+    const { path, selection } = params;
+
+    if (path === ERROR_PATH) {
+      // Throw error when fetching value with specific path
+      throw new Error('error');
+    }
+
+    const dataset = findMockEntity(path);
+    assertMockDataset(dataset);
+
+    if (path === SLOW_VALUE_PATH) {
+      // Add cancellable artificial delay
+      await this.withCancelToken(async (cancelToken) => {
+        return new Promise<void>((resolve, reject) => {
+          const timeout = setTimeout(() => resolve(), SLOW_TIMEOUT);
+
+          cancelToken.promise.then((error) => {
+            clearTimeout(timeout);
+            reject(error);
+          });
+        });
+      });
+    }
```

Comment:
This is the main change. I'm able to use the axios cancel token, since it's just a promise.